mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -521,8 +521,10 @@ object KMeans {
 /**
  * A vector with its norm for fast distance computation.
  */
-private[clustering] class VectorWithNorm(val vector: Vector, val norm: Double)
-    extends Serializable {
+private[clustering] class VectorWithNorm(
```

Comment:
I am neutral on adding weight in `VectorWithNorm`, then what about also using it in KMeans?
for example: `val zippedData: RDD[(VectorWithNorm, Double)]` => `val zippedData: RDD[VectorWithNorm]`

Reply:
I will update KMean code. Thanks!